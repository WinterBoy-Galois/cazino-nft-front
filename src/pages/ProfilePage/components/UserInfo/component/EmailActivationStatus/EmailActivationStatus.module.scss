@import '../../../../../../styles/shared';

.container {
  background-color: $email-status-dark-blue;
  border-radius: 8px;
  border: 1px solid $email-status-blue;
  padding: 8px 12px;
  color: $email-status-light-blue;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  max-width: 290px;
}

.icon {
  width: 20px;
  height: 20px;
  margin-right: 0.5rem;
}
