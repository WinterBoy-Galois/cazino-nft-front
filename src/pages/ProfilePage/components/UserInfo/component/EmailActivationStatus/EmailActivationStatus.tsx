import React from 'react';

import styles from './EmailActivationStatus.module.scss';
import AccountVerified from '../../../../../../components/icons/AccountVerified';
import AccountUnverified from '../../../../../../components/icons/AccountUnverified';

interface IProps {
  isActivated: boolean;
  email: string;
  className?: string;
}

const EmailActivationStatus: React.FC<IProps> = ({ email, className = '', isActivated }) => {
  return (
    <div className={`${styles.container} ${className}`}>
      {isActivated ? (
        <AccountVerified className={styles.icon} />
      ) : (
        <AccountUnverified className={styles.icon} />
      )}
      <span className="truncate">{email}</span>
    </div>
  );
};

export default EmailActivationStatus;
