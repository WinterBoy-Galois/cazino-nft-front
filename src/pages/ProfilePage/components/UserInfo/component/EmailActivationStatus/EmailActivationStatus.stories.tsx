import React from 'react';
import { text, boolean } from '@storybook/addon-knobs';

import EmailActivationStatus from '.';

export default {
  title: 'Components/EmailActivationStatus',
  component: EmailActivationStatus,
};

const data = {
  email: 'test@test.de',
};

export const Default = () => (
  <EmailActivationStatus
    email={text('Email', data.email)}
    isActivated={boolean('IsActivated', false)}
  />
);
