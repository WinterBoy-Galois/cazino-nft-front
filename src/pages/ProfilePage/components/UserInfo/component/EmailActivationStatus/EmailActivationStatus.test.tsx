import React from 'react';
import { render } from '@testing-library/react';

import EmailActivationStatus from '.';

describe('EmailActivationStatus', () => {
  it('should match snapshot', () => {
    // Arrange

    // Act
    const container = render(<EmailActivationStatus isActivated={true} email={'test@test.de'} />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
