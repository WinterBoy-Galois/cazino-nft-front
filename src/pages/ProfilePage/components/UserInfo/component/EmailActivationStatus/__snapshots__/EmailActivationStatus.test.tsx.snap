// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailActivationStatus should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container "
      >
        <svg
          class="icon"
          viewBox="0 0 23.13 23.13"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <g>
              <g>
                <path
                  class="fill"
                  d="M11.57 23.13a11.57 11.57 0 110-23.13 11.53 11.53 0 014.7 1 1.5 1.5 0 01-1.22 2.74A8.5 8.5 0 0011.57 3a8.57 8.57 0 000 17.13 8.55 8.55 0 008.56-8.56v-.92a1.5 1.5 0 013 0v.92a11.56 11.56 0 01-11.56 11.56z"
                />
                <path
                  class="fill"
                  d="M11.57 15.09a1.46 1.46 0 01-1.06-.44l-3-3a1.49 1.49 0 010-2.12 1.51 1.51 0 012.12 0l2 2 9-9a1.5 1.5 0 012.07 2.05L12.63 14.65a1.46 1.46 0 01-1.06.44z"
                />
              </g>
            </g>
          </g>
        </svg>
        <span
          class="truncate"
        >
          test@test.de
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container "
    >
      <svg
        class="icon"
        viewBox="0 0 23.13 23.13"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <g>
            <g>
              <path
                class="fill"
                d="M11.57 23.13a11.57 11.57 0 110-23.13 11.53 11.53 0 014.7 1 1.5 1.5 0 01-1.22 2.74A8.5 8.5 0 0011.57 3a8.57 8.57 0 000 17.13 8.55 8.55 0 008.56-8.56v-.92a1.5 1.5 0 013 0v.92a11.56 11.56 0 01-11.56 11.56z"
              />
              <path
                class="fill"
                d="M11.57 15.09a1.46 1.46 0 01-1.06-.44l-3-3a1.49 1.49 0 010-2.12 1.51 1.51 0 012.12 0l2 2 9-9a1.5 1.5 0 012.07 2.05L12.63 14.65a1.46 1.46 0 01-1.06.44z"
              />
            </g>
          </g>
        </g>
      </svg>
      <span
        class="truncate"
      >
        test@test.de
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
